### PR TITLE
storage: replace standard gzip with klauspost/compress to increase effiency

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43128

Problem Summary:

### What is changed and how it works?
replace standard gzip with klauspost/compress to increase effiency

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
 After this lightning's speed from 2x no compressed to almost the same with no compressed data.

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/writer.go [hunk 1]
```diff
@@ -2,11 +2,11 @@ package storage
 
 import (
 	"bytes"
-	"compress/gzip"
 	"context"
 	"io"
 
 	"github.com/golang/snappy"
```
Comment (gozssky) [inline br/pkg/storage/writer.go:8]: How about replacing snappy as well?

Comment (hawkingrei) [inline br/pkg/storage/writer.go:8]: Please run ```make bazel_prepare```.

Comment (lichunzhu) [reply br/pkg/storage/writer.go]: addressed in https://github.com/pingcap/tidb/pull/43129/commits/607d490095ddb176d3ec054cf4f9af0a6acc7efa

Comment (lichunzhu) [reply br/pkg/storage/writer.go]: addressed in https://github.com/pingcap/tidb/pull/43129/commits/607d490095ddb176d3ec054cf4f9af0a6acc7efa

**Review by lance6716 — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

**Pull request merged**